File: IT/Programmazione funzionale/Esercizi/Python/esercizio11.py
# file esercizio11.py

from __future__ import print_function
import math

def psq(n):
	m = round(math.sqrt(n))
	return True if (m*m) == n else False

ps = list(filter(psq, [n for n in range(1, 100+1)]))

print(ps)

File: IT/2/Programmazione funzionale/Esercizi/Java/Esercizio11.java
/* file Esercizio11.java */

import java.util.List;
import java.util.stream.IntStream;
import java.util.stream.Collectors;

public class Esercizio11 {
	private static boolean psq(int n) {
		int m = Math.round((int)Math.sqrt(n));
		return (m*m) == n ? true : false;
	}

	public static void main(String [] args) {
		List<Integer> ps = IntStream.rangeClosed(1, 100).filter(n -> psq(n)).boxed().collect(Collectors.toList());
		System.out.println(ps);
	}
}

